Improve aspect on large screens

druc opened the pull request:
For #26

@samuelmeuli I tried increasing the font for larger screens but it ended up looking bad - I like the 16px  version way better. I still believe the paragraph spacing should be at least one line though.

Let me know what you think.

Thanks again for building this. It's really nice to write in it! ❤️

Commit:
Improve aspect on large screens

Restore box shadow. Add editor-buttons-width scss variable

Responsive editor adjustment

Remove unneeded editor-scrollable max width constraint

Co-Authored-By: Samuel Meuli <me@samuelmeuli.com>

Typography and increase  editor padding.

Add media query variables

Improve aspect on large screens

Restore box shadow. Add editor-buttons-width scss variable

Merge

Remove unneeded editor-scrollable max width constraint

Co-Authored-By: Samuel Meuli <me@samuelmeuli.com>

Update src/renderer/assets/styles/components/_editor.scss

Co-Authored-By: Samuel Meuli <me@samuelmeuli.com>

samuelmeuli commented:
I like it, definitely an improvement! I'll test the changes on a larger screen tomorrow.

How did you arrive at the values for the media query breakpoints? It might be nice to combine some of them into variables.

samuelmeuli commented on src/renderer/assets/styles/base/_variables.scss:
```diff
@@ -51,7 +51,8 @@ $spacing-rel-large: 6vw;
 $border-radius-input: 5px;
 $box-shadow: 0.5px 0.5px 1px 1px rgba(0, 0, 0, 0.18);
-$box-shadow-inset: inset 0.5px 0.5px 1px 1px rgba(0, 0, 0, 0.13);
+$box-shadow-inset: 0 1px 3px 0 rgba(0, 0, 0, 0.1), 0 1px 2px 0 rgba(0, 0, 0, 0.06);
```
In my opinion, input fields should have an inset shadow, this makes it clear to the user that they're editable.

druc replied:
Yup. I've only added this so the editor could pop up a bit more. I forgot about the inputs.

samuelmeuli commented on src/renderer/assets/styles/components/_editor.scss:
```diff
@@ -9,9 +10,14 @@
 	display: grid;
 	grid-template-rows: auto auto 1fr;
 	width: 100%;
+	max-width: $max-editor-width;
```
This line isn't needed.
```suggestion
```

samuelmeuli commented on src/renderer/assets/styles/components/_editor.scss:
```diff
@@ -9,9 +10,14 @@
+	padding: $spacing-large (2.25 * $spacing-large) $spacing-large $spacing-large;
 	overflow-y: auto;
+
+	@media (min-width: 1224px) {
```
Let's use the actual size of the editor toolbar for the calculations here:

```scss
// Editor padding (width and margin of editor buttons are added to right padding)
$editor-buttons-width: $calendar-cell-size + $spacing-medium;
@media (max-width: 1224px) {
	padding: $spacing-large ($spacing-large + $editor-buttons-width) $spacing-large $spacing-large;
}
@media (min-width: 1225px) {
	padding: (2 * $spacing-large) (2 * $spacing-large + $editor-buttons-width) (2 * $spacing-large)
		(2 * $spacing-large);
}
```

samuelmeuli commented:
And regarding the paragraph spacing: How about setting `line-height: 1.9` and applying `margin-bottom: 1em` to `.editor-text-wrapper .public-DraftStyleDefault-block` on larger screens?

druc commented:
Now that we constrained the editor from going full width I don't see any reason to increase the `line-height` or `margin-bottom` **only** on larger screens.

Whenever I'm undecided about typography I usually use this tool: https://grtcalculator.com/
Even though I don't agree with it every time, it does provide a great start.

I'd go with the 1.5em spacing, but I'm a sucker for large spaces between paragraphs, so I understand other people might having their content more condensed.

Should we go with `line-height: 1.7em`, `margin-bottom: 1em`?

PS: regarding the media query 1224px breakpoint, I just needed something larger than 1200px, so I googled the most used media queries. Apparently 1224px is used for desktops and laptops 🤷‍♂.

10px margin
<img width="1284" alt="10px" src="https://user-images.githubusercontent.com/13726728/60418560-e4d07f80-9beb-11e9-9dae-f60f354743fd.png">

1em margin
<img width="1282" alt="1em" src="https://user-images.githubusercontent.com/13726728/60418558-e4d07f80-9beb-11e9-8048-e92a379f9854.png">

1.5em margin
<img width="1284" alt="1 5em" src="https://user-images.githubusercontent.com/13726728/60418557-e437e900-9beb-11e9-8e77-53a1bdd5f262.png">

samuelmeuli commented on src/renderer/assets/styles/components/_editor.scss:
```diff
@@ -10,8 +11,17 @@
 	width: 100%;
 	height: 100%;
-	padding: $spacing-rel-small (2 * $spacing-rel-small) $spacing-rel-small $spacing-rel-small;
+	padding: $spacing-large (2.25 * $spacing-large) $spacing-large $spacing-large;
```
```suggestion
```

samuelmeuli commented:
That's a cool website, thanks for the tip!

Ok, let's do the line/paragraph spacing like you suggested. Let's use `margin-top: 1em` though and remove the bottom margin from the title to make the spacing consistent.

Also, because of that change, I think we should increase the editor padding on small screens slightly again (maybe to `1.25 * $spacing-large` or what we had before), otherwise the paragraph spacing is larger than the padding, which looks a bit odd to me.

Finally, if you could extract the break points into variables, I'll be happy to merge this afterwards. Sorry for the nitpicking 😄

Commit:
Merge branch 'master' into lg-view

druc commented:
- added `margin-top: 1em`
- removed `margin-bottom` from title
- increased padding on small screens
- extracted the media query values into `$screen-size` variables. 
- increased the media query for one-column view. Anything smaller than 1000px looks cramped anyway.

We now have: 1024px for tablets, 1224px for desktops/laptops and 1824px for larger screens.

Don't worry about nitpicking. As long as the app gets better, I don't mind tweaking stuff around.

samuelmeuli commented:
Awesome, thank you! :)
